### PR TITLE
Replace localized card discount reporting, moving it to the card. Add Moon Tether.

### PR DESCRIPTION
**Pull request by kberg**
image
image

### CODE REVIEW

File: src/models/CardModel.ts [hunk 1]
```diff
@@ -3,12 +3,14 @@ import {CardType} from '../cards/CardType';
...
 import {Units} from '../Units';
 import {CardName} from '../CardName';
+import {CardDiscount} from '../cards/ICard';
 
 export interface CardModel {
     name: CardName;
     resources: number | undefined;
     resourceType: ResourceType | undefined;
     calculatedCost?: number;
+    discount?: CardDiscount,
```
Comment (bafolts) [inline src/models/CardModel.ts:13]: This makes sense to me to place on `CardMetadata`. Placing the `discount` on `CardMetadata` will also help with memory per card.

File: src/cards/base/AntiGravityTechnology.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import {CardName} from '../../CardName';
 import {CardRenderer} from '../render/CardRenderer';
 import {CardRequirements} from '../CardRequirements';
 
+export const ALL = 'all';
```
Comment (bafolts) [inline src/cards/base/AntiGravityTechnology.ts:9]: I don't think this is used anywhere. Probably leftover from when taking a different approach. Or I am missing something.

**Review by bafolts — CHANGES_REQUESTED** [review]
Separating the discount mapping from `PlayerTags` is good change. The added `cardDiscount` property will either have to use the static memory pattern or be placed on the `metadata` property. It feels like it is metadata property to me so would make sense to place on `metadata`.

File: src/cards/base/AntiGravityTechnology.ts (hunk 2 above)
Comment (kberg) [reply src/cards/base/AntiGravityTechnology.ts:9]: A, removing

File: src/models/CardModel.ts (hunk 1 above)
Comment (kberg) [reply src/models/CardModel.ts:13]: How about making that a follow-up change?

**Comment by kberg** [comment]
> Separating the discount mapping from `PlayerTags` is good change. The added `cardDiscount` property will either have to use the static memory pattern or be placed on the `metadata` property. It feels like it is metadata property to me so would make sense to place on `metadata`.

I'm ... not so sure. But OK.

**Comment by kberg** [comment]
> > Separating the discount mapping from `PlayerTags` is good change. The added `cardDiscount` property will either have to use the static memory pattern or be placed on the `metadata` property. It feels like it is metadata property to me so would make sense to place on `metadata`.
> 
> I'm ... not so sure. But OK.

Do you mean CardMetadata? CardMetadata has mostly been used for rendering. I'd like to keep those separate, and prefer to use the static memory pattern (and replace calls to getCardDiscount where reasonable.)

**Review by bafolts — APPROVED** [review]
Using the static memory pattern would also be fine. Doing in a separate pull request is fine.

**Comment by kberg** [comment]
> Using the static memory pattern would also be fine. Doing in a separate pull request is fine.

Already done.

**Pull request merged**